python/pyspark/mllib/linalg.py

```diff
@@ -1152,9 +1156,385 @@ def sparse(numRows, numCols, colPtrs, rowIndices, values):
         return SparseMatrix(numRows, numCols, colPtrs, rowIndices, values)
 
 
+class DistributedMatrix(object):
+    """Represents a distributively stored matrix backed by one or more RDDs."""
+    def numRows(self):
```

Comment:
If you would like to represent this as an abstract class, it might be better to use `ABCMeta` and `abstract method` from `abc` module. Check out `pyspark/ml/pipeline.py`

One quick advantage that I can recall from memory is that an error is raised during object creation rather than when the methods are called as it does now. (I think)